go/mysql/connection.go

```diff
@@ -42,13 +50,39 @@ func (this *ConnectionConfig) Duplicate() *ConnectionConfig {
 }
 
 func (this *ConnectionConfig) String() string {
-	return fmt.Sprintf("%s, user=%s", this.Key.DisplayString(), this.User)
+	return fmt.Sprintf("%s, user=%s, usingTLS=%t", this.Key.DisplayString(), this.User, this.tlsConfig != nil)
 }
 
 func (this *ConnectionConfig) Equals(other *ConnectionConfig) bool {
 	return this.Key.Equals(&other.Key) || this.ImpliedKey.Equals(other.ImpliedKey)
 }
 
+func (this *ConnectionConfig) UseTLS(caCertificatePath string) error {
+	skipVerify := caCertificatePath == ""
+	var rootCertPool *x509.CertPool
+	if !skipVerify {
+		rootCertPool = x509.NewCertPool()
+		pem, err := ioutil.ReadFile(caCertificatePath)
+		if err != nil {
+			return err
+		}
+		if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+			return errors.New("could not add ca certificate to cert pool")
+		}
+	}
```

Comment:
@brandonbodnar-wk Can you tweak the logic here so that if it uses ssl, it verifies the cert by default?

Reply:
(Like how curl only skips cert verification if you pass a `--insecure` flag.)

Reply:
cc @ericklaus-wf

Reply:
@matthewbalvanz-wf That is not the default behavior for mysql. This (calling `--ssl` in on the command line) attempts to emulate the behavior of `--ssl-mode=REQUIRED` in [5.7](https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-options.html#option_general_ssl-mode), which is the same as `--enable-ssl` for [mysql 5.6](https://dev.mysql.com/doc/refman/5.7/en/encrypted-connection-options.html#option_general_ssl). That setting only requires that the connection by encrypted with no guarantees about the certificate provided by the server is from a valid CA (self-signed certificates are okay).

Those (like us) who want additional guarantees that the cert is signed by a specific CA authority, can use the additional `--ssl-ca` flag to behave like `--ssl-mode=VERIFY_CA` from the mysql command line client.

Reply:
The `caCertificatePath` is set using that flag.

Reply:
Hmm I see what you're saying @brandonbodnar-wk , but it seems best to make that `--ssl` flag emulate `VERIFY_CA` instead, unless an `--ssl-insecure` flag is specifically set. The rest of the infosec team agrees that having ssl without verification by default is not a good idea. Here's a PR that achieves what I'm thinking.

https://github.com/Workiva/gh-ost/pull/4

Can you check that out and merge it if it looks good?